### PR TITLE
[GEOS-9561] Add rest parameter to update native bbox when harvesting …

### PR DESCRIPTION
**Pull request by taba90**
…or deleting granules from image mosaic

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [x] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by taba90** [comment]
jira ticket https://osgeo-org.atlassian.net/browse/GEOS-9561

File: doc/en/api/1.0.0/coveragestores.yaml [hunk 1]
```diff
@@ -286,6 +286,12 @@ paths:
           required: false
           description: The filename parameter specifies the target file name for a file that needs to harvested as part of a mosaic. This is important to avoid clashes and to make sure the right dimension values are available in the name for multidimensional mosaics to work. Only used if method="file".
           type: string
+        - name: updateBBox
+          in: query
+          required: false
+          description: The parameter allows, when valorized as "true", to trigger the native bbox update of the layer configuration according
```
Comment (aaime) [inline doc/en/api/1.0.0/coveragestores.yaml:292]:
```suggestion
          description: When set to "true", triggers re-calculation of the layer native bbox. Defaults to "false".
```

File: doc/en/api/1.0.0/structuredcoverages.yaml [hunk 2]
```diff
@@ -402,6 +402,11 @@ paths:
           required: false
           type: string
           minimum: 0
+        - name: updateBBox
+          in: query
+          required: false
+          description: The parameter allows, when valorized as "true", to trigger the native bbox update of the layer configuration according
```
Comment (aaime) [inline doc/en/api/1.0.0/structuredcoverages.yaml:408]: See comment above.

File: doc/en/api/1.0.0/structuredcoverages.yaml [hunk 3]
```diff
@@ -589,7 +594,12 @@ paths:
             Finally, when set to "all" both data and auxiliary files are removed.
           required: false
           type: string
-          minimum: 0         
+          minimum: 0  
+        - name: updateBBox
+          in: query
+          required: false
+          description: The parameter allows, when valorized as "true", to trigger the native bbox update of the layer configuration according
```
Comment (aaime) [inline doc/en/api/1.0.0/structuredcoverages.yaml:601]: See comment above.

File: src/restconfig/src/main/java/org/geoserver/rest/catalog/MosaicInfoBBoxHandler.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.logging.Logger;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.CoverageInfo;
+import org.geoserver.catalog.CoverageStoreInfo;
+import org.geotools.coverage.grid.io.StructuredGridCoverage2DReader;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.geotools.util.logging.Logging;
+
+/**
+ * This class aims to provide methods to update mosaic native bounding box at the coverage info
```
Comment (aaime) [inline src/restconfig/src/main/java/org/geoserver/rest/catalog/MosaicInfoBBoxHandler.java:19]:
```suggestion
 * This class provides methods to update a mosaic native bounding box at the coverage info
```

File: src/restconfig/src/test/java/org/geoserver/rest/catalog/CoverageStoreFileUploadTest.java [hunk 5]
```diff
@@ -576,4 +572,108 @@ private void configureCoverageInfo(
...
+                storeInfo, (current, old) -> assertNotEquals(current, old), catalog);
+        CoverageInfo coverage = getCatalog().getResourceByName("bboxtest", CoverageInfo.class);
+        if (coverage != null) {
+            removeStore(
+                    coverage.getStore().getWorkspace().getName(), coverage.getStore().getName());
+        }
+    }
+
+    @Test
+    public void testUpdateBBoxTrueParameterPOST() throws Exception {
```
Comment (aaime) [inline src/restconfig/src/test/java/org/geoserver/rest/catalog/CoverageStoreFileUploadTest.java:610]: Good to see tests for harvest and re-harvest. The delete path was modified too, is there a test for it?

File: doc/en/api/1.0.0/coveragestores.yaml (hunk 1 above)
Comment (taba90) [reply doc/en/api/1.0.0/coveragestores.yaml]: fixed

File: doc/en/api/1.0.0/structuredcoverages.yaml (hunk 2 above)
Comment (taba90) [reply doc/en/api/1.0.0/structuredcoverages.yaml]: fixed

File: doc/en/api/1.0.0/structuredcoverages.yaml (hunk 3 above)
Comment (taba90) [reply doc/en/api/1.0.0/structuredcoverages.yaml]: fixed

File: src/restconfig/src/main/java/org/geoserver/rest/catalog/MosaicInfoBBoxHandler.java (hunk 4 above)
Comment (taba90) [reply src/restconfig/src/main/java/org/geoserver/rest/catalog/MosaicInfoBBoxHandler.java]: fixed

File: src/restconfig/src/test/java/org/geoserver/rest/catalog/CoverageStoreFileUploadTest.java (hunk 5 above)
Comment (taba90) [reply src/restconfig/src/test/java/org/geoserver/rest/catalog/CoverageStoreFileUploadTest.java:610]: yes those are in the StructuredCoverageStoresTest class https://github.com/geoserver/geoserver/pull/4156/commits/9c3df0dec3079791dd5f99101180e5ffe5e92bb9#diff-12372cf7938b4aed6e65eba2d824237aR871

**Comment by taba90** [comment]
Thanks for your review @aaime feedbacks have been applied. I've also added a commit to fix MacOs build failings. This build failings are due unrelated tests not caused by gs-restconfig module that indeed passed all: 

[INFO] Web Map Service Module ............................. SUCCESS [04:20 min]
[INFO] KML support for GeoServer .......................... SUCCESS [ 28.799 s]
[INFO] gs-rest ............................................ SUCCESS [ 12.577 s]
[INFO] GeoWebCache (GWC) Module ........................... FAILURE [02:05 min]
[INFO] gs-restconfig ...................................... SUCCESS [02:37 min]
[INFO] gs-restconfig-wcs .................................. SUCCESS [ 14.317 s]
[INFO] gs-restconfig-wfs .................................. SUCCESS [ 19.548 s]
[INFO] gs-restconfig-wms .................................. SUCCESS [ 13.566 s]

**Comment by nmco** [comment]
Looks like all review comments have been applied, merging.

**Pull request merged**